src/cql/upstream-external/C-IDAS_LogicLibrary.cql

```diff
@@ -0,0 +1,353 @@
+    else if score is null then 0
+    else score as Integer
+
+// Question 13
+define CIDASScoreQuestion13:
+  (getQuestionScore(IndexOfQuestion.thirteen, question13LinkId)) score
+  return
+    if score = 'a' then 0
+    else if score is null then 0
+    else score as Integer
+
+// Question 14
+define CIDASScoreQuestion14:
+  (getQuestionScore(IndexOfQuestion.fourteen, question14LinkId)) score
+  return
+    if score = 'a' then 0
+    else if score is null then 0
+    else score as Integer
+
+// Question 15
+define CIDASScoreQuestion15:
+  (getQuestionScore(IndexOfQuestion.fifteen, question15LinkId)) score
+  return
+    if score = 'a' then 0
+    else if score is null then 0
+    else score as Integer
+
+// Question 16
+define CIDASScoreQuestion16:
+  (getQuestionScore(IndexOfQuestion.sixteen, question16LinkId)) score
+  return
+    if score = 'a' then 0
+    else if score is null then 0
+    else score as Integer
+
+// Question 17
+define CIDASScoreQuestion17:
+  (getQuestionScore(IndexOfQuestion.seventeen, question17LinkId)) score
+  return
+    if score = 'a' then 0
+    else if score is null then 0
+    else score as Integer
+
+// Question 18
+define CIDASScoreQuestion18:
+  (getQuestionScore(IndexOfQuestion.eighteen, question18LinkId)) score
+  return
+    if score = 'a' then 0
+    else if score is null then 0
+    else score as Integer
+
+// Should the score for the Full C-IDAS be displayed?
+define DisplayScoreFullCIDAS:
+   (
+    (getCurrentResponseByQuestionId(question1LinkId)) is not null or 
+    (getCurrentResponseByQuestionId(question2LinkId)) is not null or
+    (getCurrentResponseByQuestionId(question3LinkId)) is not null or
+    (getCurrentResponseByQuestionId(question4LinkId)) is not null or 
+    (getCurrentResponseByQuestionId(question5LinkId)) is not null or
+    (getCurrentResponseByQuestionId(question6LinkId)) is not null or
+    (getCurrentResponseByQuestionId(question7LinkId)) is not null or
+    (getCurrentResponseByQuestionId(question8LinkId)) is not null or
+    (getCurrentResponseByQuestionId(question9LinkId)) is not null or
+    (getCurrentResponseByQuestionId(question10LinkId)) is not null or
+    (getCurrentResponseByQuestionId(question11LinkId)) is not null or
+    (getCurrentResponseByQuestionId(question12LinkId)) is not null or
+    (getCurrentResponseByQuestionId(question13LinkId)) is not null or
+    (getCurrentResponseByQuestionId(question14LinkId)) is not null or
+    (getCurrentResponseByQuestionId(question15LinkId)) is not null or
+    (getCurrentResponseByQuestionId(question16LinkId)) is not null or
+    (getCurrentResponseByQuestionId(question17LinkId)) is not null or
+    (getCurrentResponseByQuestionId(question18LinkId)) is not null
+
+   ) // at least one response for all the questions
+    and DisplayScreeningScores // Input parameter which allows score display to be disabled
+
+// Calculate the score for the Full C-IDAS
+// sum up all numeric values for all the answers
+define ScoreFullCIDAS:
```

Comment:
As we discussed just now, much of the scoring code is repeated in https://github.com/uwcirg/patient-summary/tree/feature/initial-patient-summary , would be good to move this into a Library. We also discussed sharing a Library w/ COSRI, and the fact that COSRI's cql-execution would need to be updated (currently uses 1.3.7, while this repo uses 2.4.x)